src/node.rs

```diff
@@ -118,7 +120,69 @@ impl Node {
 
         mem::replace(&mut self.current_candidate_blocks,
                      compute_current_candidate_blocks(potentially_current));
-        self.current_blocks = compute_current_blocks(&self.current_candidate_blocks);
+
+        mem::replace(&mut self.current_blocks,
+                     compute_current_blocks(&self.current_candidate_blocks))
+
+    }
+
+    fn merge_messages(&self, prev_current_blocks: &CurrentBlocks)
+        -> Vec<Message> {
+        let mut messages = vec![];
+        let new_current_blocks = &self.current_blocks - prev_current_blocks;
+
+        for our_new_block in section_blocks(&new_current_blocks, self.our_name) {
+            for our_prev_block in section_blocks(&prev_current_blocks, self.our_name) {
+                // Merge case.
+                if let Some(sibling_pfx) = our_prev_block.prefix.sibling() {
+                    if our_new_block.prefix == our_prev_block.prefix.popped() {
+                        // In the case of a merge, send our sibling's history to all the
+                        // sections we are connected to but they are not.
+                        // FIXME: need to send history for all descendents of sibling prefix?
+                        // i.e. if we are 0 and want to merge to 1, but other
+                        // sections are 11, 100, 101.
+                        let sibling_block = prev_current_blocks.iter()
+                            .find(|block| block.prefix == sibling_pfx)
```

Comment:
Fixing the `FIXME` should just be changing this one line, shouldn't it? I think it would be good to include that change.